internal/service/eks/find.go

```diff
@@ -68,6 +68,56 @@ func FindAddonUpdateByClusterNameAddonNameAndID(ctx context.Context, conn *eks.E
 	return output.Update, nil
 }
 
+func FindAddonVersionByAddonNameAndKubernetesVersion(ctx context.Context, conn *eks.EKS, addonName, kubernetesVersion string, latest bool) (*eks.AddonVersionInfo, error) {
```

Comment:
I don't see `defaultVersion` being respected. Even with a data source filter, I think respecting `defaultVerison` should be the default behavior.

Reference:
https://docs.aws.amazon.com/eks/latest/APIReference/API_DescribeAddonVersions.html#API_DescribeAddonVersions_SeeAlso

Reply:
The `most_recent` (optional bool) argument to the datasource is false by default. This activates the search across all elements for the `defaultVersion=true` in the version.Compatibilities loop.